pkg/controller/job/job_controller.go

```diff
@@ -1937,6 +1916,17 @@ func errorFromChannel(errCh <-chan error) error {
 	return nil
 }
 
+func parallelizeUntil(ctx context.Context, pieces int, doWorkPiece func(i int) error) error {
+	errCh := make(chan error, pieces)
+	workqueue.ParallelizeUntil(ctx, pieces, pieces, func(i int) {
```

Comment:
That is a good point. I think that is the tradeoff.
If we do not limit the number of workers, the operation throughput will increase as long as the APIServer performances. But, in the resource-limited cluster, it will cause APF issues.

In anycase, I think that we need to decide the number of workers. So, could we track it separately?

Note that the number of workers is the number of pods in the current implementation (master branch).

Reply:
Yup, we can track that separately.

Reply:
@atiratree I created tracking issue. Let's discuss it there.

https://github.com/kubernetes/kubernetes/issues/128625